src/main/cpp/hw/DragonSparkFlex.cpp

```diff
@@ -187,12 +203,15 @@ bool DragonSparkFlex::IsForwardLimitSwitchClosed()
 
 bool DragonSparkFlex::IsReverseLimitSwitchClosed()
 {
+    if (m_spark->GetReverseLimitSwitch(m_reverseType).Get())
+        m_encoder.SetPosition(0);
     return m_spark->GetReverseLimitSwitch(m_reverseType).Get();
 }
 
 void DragonSparkFlex::EnableDisableLimitSwitches(bool enable)
 {
-    m_sparkLimitSwitch->EnableLimitSwitch(enable);
+    SparkLimitSwitch limitSwitch = m_spark->GetForwardLimitSwitch(m_forwardType);
```

Comment:
This can use m_forwardLimitSwitch and m_reverseLimitSwitch.  The function name makes me think that both limit switches should be enabled/disabled, not just one.

Reply:
Agreed, resolved